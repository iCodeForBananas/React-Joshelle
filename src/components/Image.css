.Image-img {
  max-width: 300px;
  max-height: 300px;
  position: absolute;
}
