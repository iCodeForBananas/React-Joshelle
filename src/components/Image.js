// @flow
import React from "react";
import "./Image.css";

type Props = {
  src: string,
  top: number,
  left: number
};

const Image = function({ src, left, top }: Props) {
  return <img style={{ marginLeft: left + "px", marginTop: top + "px" }} className="Image-img" src={src} alt="" />;
};

export default Image;
