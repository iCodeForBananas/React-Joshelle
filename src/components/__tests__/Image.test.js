import React from "react";
import Image from "../Image";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<Image src="IMAGE_PATH_HERE" top="100" left="100" />).toJSON();
  expect(tree).toMatchSnapshot();
});
