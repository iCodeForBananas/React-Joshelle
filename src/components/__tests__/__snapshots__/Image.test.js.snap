// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<img
  alt=""
  className="Image-img Image-spin"
  src="IMAGE_PATH_HERE"
  style={
    Object {
      "marginLeft": "100px",
      "marginTop": "100px",
    }
  }
/>
`;
